app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,34 @@
+class CommentsController < ApplicationController
+  before_action :require_login
+
+  def new
+    @comment = Comment.new(commentable_id: params[:commentable][:commentable_id], commentable_type: params[:commentable][:commentable_type])
+  end
+
+  def create
+    comment = Comment.new(comments_params)
```

Comment:
You have >1 create routes that hit this action
```
   question_comments POST   /questions/:question_id/comments(.:format)     comments#create
    comment_comments POST   /comments/:comment_id/comments(.:format)       comments#create
                     POST   /comments(.:format)                            comments#create
```

So you could infer the commentable type from whether those params exist